tests/modules/blockedclient.c

```diff
@@ -285,6 +285,11 @@ int do_rm_call_async(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
 
     format[format_len++] = 'E';
     format[format_len++] = 'v';
+    if (strcasecmp(invoked_cmd, "do_rm_call_async_no_replicate") != 0) {
+        /* Notice, without the '!' flag we will have inconsistency between master and replica.
+         * This is used only to check '!' flag correctness on blocked commands. */
+        format[format_len++] = '!';
+    }
```

Comment:
in other test units (IIRC misc.c) we abandoned this approach of matching the function name, and instead passed the flags and first argument to the function (before the arguments it passes on to RM_Call's argv).
this makes the tests more flexible in the sense that you control everything from the tcl file and don't need to add features to the c file every time.
on the other hand, we can argue it's less readable, unless you consider these RM_Call flags a well known thing, and everyone know what `!` and `K` does.

so let's consider converting the several usage of this approach in this file, let me know what you think,

Reply:
Sure, I can match to this approach but notice that then the `tcl` tests will have those flags pass on each call so not sure it will be cleaner..

Reply:
we can pass only the explicit ones, and keep some implicit.
also, we can make the `!` implicit, unless `?` is added, but that's a little bit confusing.
what we have now is also ok, but it means we need to add code to the C file every time we need a new flag.
i'm looking for something that's easiest for the reader..
the key point is not to force the reader of the tcl code to go look at the C code when reading each individual TCL test, so that's already covered.
if we can clean the C code without compromising that, let's do it, otherwise, we can skip

Reply:
> also, we can make the ! implicit, unless ? is added, but that's a little bit confusing.

Yes I think it will really be confusing. I can make the command get the flags a the first argument, but then we will have to pass it on all the tests. Or we can stay with the current version. Let me know what you prefer.

Reply:
let's keep the current one. i suppose in this case it's more explicit and easier to read.